lib/full-site-editing/edit-site-export.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * REST endpoint for exporting the contents of the Edit Site Page editor.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Output a ZIP file with an export of the current templates
+ * and template parts from the site editor, and close the connection.
+ */
+function gutenberg_edit_site_export() {
+	// Create ZIP file and directories.
+	$filename = tempnam( get_temp_dir(), 'edit-site-export' );
+	if ( ! class_exists( 'ZipArchive' ) ) {
+		return new WP_Error( 'Zip Export not supported.' );
+	}
+	$zip = new ZipArchive();
+	$zip->open( $filename, ZipArchive::OVERWRITE );
+	$zip->addEmptyDir( 'theme' );
+	$zip->addEmptyDir( 'theme/block-templates' );
+	$zip->addEmptyDir( 'theme/block-template-parts' );
```

Comment:
```suggestion
	$theme = wp_get_theme()->get_stylesheet();
	$zip->addEmptyDir( $theme );
	$zip->addEmptyDir( "$theme/block-templates" );
	$zip->addEmptyDir( "$theme/block-template-parts" );
```

maybe?

Reply:
I'm not familiar with `ZipArchive`, but @ockham suggestion might not work if `addEmptyDir` is not able to create nested directories in a single call.
(Which, I guess it's the reason we create `$zip->addEmptyDir( 'theme' );` before the templates sub-dirs

Meaning, a theme stylesheet is the theme full path relative to `wp_content/themes`.
If the theme is installed in a subdirectory (e.g. `wp_content/themes/foobar/theme`), its stylesheet would be `foobar/theme`.
Which might be an issue at `$zip->addEmptyDir( $theme );`.

Of course, if `addEmptyDir` can create the entire provided path, please disregard this comment! 😅

Reply:
I think this makes sense, Let's leave this separately, this PR is big enough.